domains/client/block-preprocessor/src/lib.rs

```diff
@@ -314,6 +301,26 @@ where
                 ));
             }
 
+            // Using one instance of runtime_api throughout the loop in order to maintain context
+            // between them.
+            // Using `check_transaction_validity_and_do_pre_dispatch` instead of `check_transaction_validity`
+            // to maintain side-effect in the storage buffer.
+            let is_legal_tx = runtime_api
+                .check_transaction_validity_and_do_pre_dispatch(
+                    at,
+                    &extrinsic,
+                    domain_block_number,
+                    at,
+                )?
+                .is_ok();
+
+            if !is_legal_tx {
+                // TODO: Generate a fraud proof for this invalid bundle
```

Comment:
This TODO can be removed since we only need to generate fraud proof if there is a mismatch of `ER::inbox_bundles`

Reply:
Plz resolve this comment